### PR TITLE
RPRBLND-1689: Call for scene update on MaterialX update

### PR DESCRIPTION
**Pull request by bnagirniak**
### PURPOSE
Updating material nodes in native Blender shader editor or MaterialX editor should be reflected in active viewport renderers.

### EFFECT OF CHANGE
Implemented updating viewport renderers during updating material nodes in native Blender shader editor and MaterialX editor.

### TECHNICAL STEPS
1. Implemented call of MaterialProperties.update() during MaterialX nodes changes and Blender Material nodes changes. Applied handlers from:
    - from depsgraph update, when material is changed in Blender shader editor
    - MX nodetree update
    - MX props update
2. MaterialProperties.update() notifies USD nodes and active viewport engines about material update, which change required USD stages with new material.
3. Made some code improvements.

### NOTES FOR REVIEWERS
There are places in code which would be good to refactor and do some optimizations. Will do this in future tasks.

### CODE REVIEW

File: src/hdusd/engine/viewport_engine.py [hunk 1]
```diff
@@ -348,15 +304,83 @@ def draw(self, context):
...
             self.notify_status("Rendering Done", "", False)
 
-    def update_render(self, scene):
-        prop = scene.hdusd.viewport
-        if self.is_gl_delegate != prop.is_gl_delegate:
-            self.is_gl_delegate = prop.is_gl_delegate
 
-            # update all scene lights since on renderer change
-            self.resync_scene_lights(scene)
+class ViewportEngineScene(ViewportEngine):
```
Comment (bsavery) [inline src/hdusd/engine/viewport_engine.py:308]: What is the purpose of this class?  Can you add some documentation string?

File: src/hdusd/mx_nodes/node_tree.py [hunk 2]
```diff
@@ -50,6 +52,17 @@ def output_node_volume(self):
         return next((node for node in self.nodes
                      if node.bl_idname == 'hdusd.MxNode_STD_volumematerial'), None)
 
+    def safe_call(self, op, *args, **kwargs):
+        """This function prevents call of self.update() during calling our function"""
+        if self._is_safe_call:
+            return op(*args, **kwargs)
```
Comment (bsavery) [inline src/hdusd/mx_nodes/node_tree.py:58]: technically couldn't the call come between the if and return?  There must be some other way to do this with a multithread lock  it just seems the logic here is strange.

File: src/hdusd/engine/viewport_engine.py (hunk 1 above)
Comment (bnagirniak) [reply src/hdusd/engine/viewport_engine.py:308]: This is Viewport engine for rendering Blender current scene. Added comment

File: src/hdusd/mx_nodes/node_tree.py (hunk 2 above)
Comment (bnagirniak) [reply src/hdusd/mx_nodes/node_tree.py:58]: Renamed to more related: safe_call -> no_update_call, _is_safe_call -> _do_update.

**Pull request merged**